.github/workflows/github-action-lint.yaml

```diff
@@ -0,0 +1,37 @@
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_call:
+
+name: Lint (GitHub Actions)
+
+jobs:
+  check-github-actions:
+    name: Check GitHub Actions
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Check for affected paths
+        id: changes
+        uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50 # v2.11.1
```

Comment:
Can we add this to the allow list? https://github.com/gravitational/github-terraform/blob/5c241b8e8b48985ba5ef4ae1b9641d0031ce2284/gravitational/repo-public.tf#L926-L936

Reply:
https://github.com/gravitational/github-terraform/pull/733